### PR TITLE
Fix strip_url() removing default port from password.

### LINKED ISSUE #7604: `strip_url()` strips :80 from the password
```
In [3]: strip_url("http://user:80@example.com:80/p", strip_credentials=False)
Out[3]: 'http://user@example.com/p'
```

### PR DESCRIPTION
**Pull request by Nishieee**
### summary
fixes `strip_url()` stripping default ports from passwords when `strip_credentials=False`.

```python
strip_url("http://user:80@example.com:80/p", strip_credentials=False)
# before: 'http://user@example.com/p'
# after:  'http://user:80@example.com/p'
```
fixes #7604

### root cause

`netloc.replace(f":{parsed_url.port}", "")` is a global substring replace and matches `:80` inside `user:password@` too.

### fix

split netloc on the rightmost `@` with `rpartition("@")` and strip the `:port` suffix from the host side only.

### changes

- `scrapy/utils/url.py` — strip default port from host only
- `tests/test_utils_url.py` — regression tests for passwords `80`, `443`, `21`
- `docs/news.rst` — bug fix entry

### test plan

```
pytest tests/test_utils_url.py
pytest tests/test_spidermiddleware_referer.py
```

edge cases verified: no credentials, non-default ports, IPv6 with and without credentials, `strip_credentials=True` path.

### CODE REVIEW

File: docs/news.rst [hunk 1]
```diff
@@ -237,6 +237,10 @@ Bug fixes
     tag when adding the ``<base>`` tag.
     (:issue:`7459`)
 
+-   Fixed :func:`scrapy.utils.url.strip_url` removing default ports from
+    passwords when ``strip_credentials=False``.
+    (:issue:`7604`)
+
```
Comment (AdrianAtZyte) [inline docs/news.rst]: Revert this change entirely, we update the release notes on new releases anyway.

File: scrapy/utils/url.py [hunk 2]
```diff
@@ -139,7 +139,11 @@ def strip_url(
             ("ftp", 21),
         }
     ):
-        netloc = netloc.replace(f":{parsed_url.port}", "")
+        port_suffix = f":{parsed_url.port}"
+        userinfo, sep, host = netloc.rpartition("@")
+        if host.endswith(port_suffix):
+            host = host[: -len(port_suffix)]
+        netloc = f"{userinfo}{sep}{host}"
```
Comment (AdrianAtZyte) [inline scrapy/utils/url.py]: What about removing the partitioning and instead using `netloc.endswith()`?

**Comment by Nishieee** [comment]
installed pre-commit locally, ran pre-commit run --all-files, and pushed the ruff removesuffix fix.

**Review by wRAR — APPROVED** [review]

**Review by AdrianAtZyte — APPROVED** [review]
